### PR TITLE
fix(card): set default background on unselected non-flat card

### PR DESCRIPTION
**Pull request by BalbinaK**
Fixes [WARP-430](https://nmp-jira.atlassian.net/browse/WARP-430)

Background color of a on-flat Card will no longer be inherited from the parent container.

Before:
Screenshot 2023-12-19 at 13 38 04

After:
Screenshot 2023-12-19 at 15 07 38

### CODE REVIEW

File: packages/card/src/component.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export function Card(props: CardProps) {
       className: classNames(props.className, {
         [ccCard.card]: true,
         [ccCard.cardShadow]: !props.flat,
-        [ccCard.cardSelected]: props.selected,
+        [props.selected ? ccCard.cardSelected : ccCard.cardUnselected]: !props.flat,
```
Comment (magnuh) [inline packages/card/src/component.tsx:23]: The flat variant would never get the selected state? 🤔

Comment (magnuh) [reply packages/card/src/component.tsx:23]: But maybe it shouldn't, but did before? 😅

Comment (BalbinaK) [reply packages/card/src/component.tsx:23]: It wouldn't? Then why 25 handles selected/unselected state for flat card? :D

Comment (magnuh) [reply packages/card/src/component.tsx:23]: Before, a selected flat card would have got both `ccCard.selected` (row 23) and `ccCardFlatSelected` (row 25)? Right? 🤷😄

**Review by magnuh — APPROVED** [review]

**Pull request merged**